### ISSUE TITLE
Syntax error when running Homebridge on RPI (new install)

### ISSUE THREAD

**Issue by alexeyovich** [nfarina/homebridge#2035]
Hi, 

I've installed Homebridge on my Raspberry PI Zero and created the config.json file, referencing my TPLink smartplug, however, when I try running Homebridge, I get the following error. 
I've had some issues getting to this point and was hoping it would work now. :(

(...)

(node:3527) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[11/10/2018, 23:20:47] There was a problem reading your config.json file.
[11/10/2018, 23:20:47] Please try pasting your config.json file here to validate it: http://jsonlint.com
[11/10/2018, 23:20:47]
/usr/local/lib/node_modules/homebridge/lib/server.js:214
    throw err;
    ^

SyntaxError: Unexpected string in JSON at position 41
    at JSON.parse (<anonymous>)
    at Server._loadConfig (/usr/local/lib/node_modules/homebridge/lib/server.js:208:19)
    at new Server (/usr/local/lib/node_modules/homebridge/lib/server.js:56:38)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:30:16)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)

(...)

I used Nano to create the file and manually typed in the appropriate lines. 
My RPI Zero is on the latest OS version (Raspbian Stretch). 

I think I've followed all the necessary steps to get Homebridge up and running, and I've already researched the error, but can't find the cause. 

### Versions
When running `homebridge --version`, I get the following: 
(node:3547) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
0.4.45

node = v8.11.1 
OS = Raspbian v9 (Stretch)
CPU architecture = ARM v61

### Configuration
Contents of my config.json file (username and pin removed)... 

(...) 

{
        "bridge": {
                "name": "Homebridge"
                "username": "xx:xx:xx:xx:xx:xx",
                "port": 51826,
                "pin": "xxx-xx-xxx"
        },

        "description": "This is an example configuration file"

        "platforms": [
                {
                        "platform" : "TplinkSmarthome",
                        "name" : "TplinkSmarthome"

                        "inUseThreshold": 0,
                        "timeout": 5
                }
        ]
}

(...)

**Comment by alexeyovich** [comment]
I should also add that I’m pretty sure that Homebridge ran, before I created the config.json file. 

Can anybody advise what the problem might be? 
Maybe something in the config file?

**Comment by calanish** [comment]
Paste the config file into https://jsonlint.com/ - it will show any parsing
errors.

I use^Wneed it way too often :)

  f

**Comment by alexeyovich** [comment]
Thanks for the suggestion. 
There was a comma missing, so I’ve added it and verified that the JSON file is now valid, however, when I run Homebridge, it doesn’t like the MAC address I’ve entered in the file. 
I’ll double check it and then will update here.

**Comment by alexeyovich** [comment]
MAC address meets the format criteria and matches that of my PI ZERO (I’ve removed the MAC address): 


From... 
~/.homebridge 

$ homebridge
(node:899) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
/usr/local/lib/node_modules/homebridge/lib/server.js:222
      throw new Error('Not a valid username: ' + username + '. Must be 6 pairs of colon-' +
      ^

Error: Not a valid username: xx:xx:xx:xx:xx:xx. Must be 6 pairs of colon-separated hexadecimal chars (A-F 0-9), like a MAC address.
    at Server._loadConfig (/usr/local/lib/node_modules/homebridge/lib/server.js:222:13)
    at new Server (/usr/local/lib/node_modules/homebridge/lib/server.js:56:38)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:30:16)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)

**Comment by iRayanKhan** [comment]
Use this to generate a MAC adress: https://www.miniwebtool.com/mac-address-generator/

**Comment by alexeyovich** [comment]
Thank you Rayan!! 

Reading the information on that page, made me think about the format of the MAC address, so I changed the alphabetical characters to upper case and that resolved the problem. 

I can now use SIRI to control my smart home bits and pieces! 

I didn’t realise that MAC addresses were case sensitive (I entered it into the config file in the same format that was reported by my PI). 

Thanks again and I hope that this helps someone else.
